include/dsn/cpp/replicated_service_app.h

```diff
@@ -165,11 +167,11 @@ namespace dsn
         };
 
         virtual ::dsn::error_code get_checkpoint(
-            int64_t start,
+            int64_t learn_start,
             int64_t local_commit,
             void*   learn_request,
             int     learn_request_size,
-            /* inout */ app_learn_state& state
+            app_learn_state& state
```

Comment:
generally, non-const reference means it is an out param.  And, why "inout" but not "out"?